.github/workflows/ci-django-api.yml

```
@@ -0,0 +1,237 @@
+name: ci-django-api
+on:
+  workflow_call:
+    inputs:
+      # required parameters
+      python-version:
+        description: 'Python version to use'
+        required: true
+        type: string
+      database-type:
+        description: 'Database management system, postgres or postgis'
+        required: true
+        type: string
+      # optional parameters
+      database-image:
+        description: 'Database image name'
+        required: false
+        default: 'postgis/postgis'
+        type: string
+      database-image-version:
+        description: 'Database image version'
+        required: false
+        default: '14-3.2-alpine'
+        type: string
+      pytest-parameters:
+        description: 'Pytest parameters'
+        required: false
+        default: '--doctest-modules'
+        type: string
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+env:
+  SECRET_KEY: topsecret123
+  POSTGRES_USER: db_user
+  POSTGRES_PASSWORD: db_password
+  POSTGRES_DB: test
+  DATABASE_URL: "${{ inputs.database-type }}://db_user:db_password@localhost/test"  
+  FIELD_ENCRYPTION_KEYS: abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  pytest:
+    name: Testing - pytest
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: ${{ inputs.database-image }}:${{ inputs.database-image-version }}
+        env:
+          # Note: not redundant, parent env is not applied to service
+          POSTGRES_USER: db_user
+          POSTGRES_PASSWORD: db_password
+          POSTGRES_DB: test
+          FIELD_ENCRYPTION_KEYS: abcdef0123456789abcdef0123456789abcdef0123456789abcdef0123456789
+        ports:
+          - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
```

There is also version 4 now.
```suggestion
        uses: actions/checkout@v4
```

---

muutettu